packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -1,31 +1,47 @@
 import * as React from 'react';
 import { View } from 'react-native';
-import { ShadowProps } from './Shadow.types';
-
-export const Shadow: React.FunctionComponent<ShadowProps> = (props: ShadowProps) => {
-  const noOffset = { width: 0, height: 0 };
-  const neutralShadowAmbient = { shadowOffset: noOffset, shadowOpacity: 0.12, shadowRadius: 2 };
-  const neutralShadowAmbientDarker = { shadowOffset: noOffset, shadowOpacity: 0.2, shadowRadius: 8 };
-  const ambientShadow = {
-    '2': neutralShadowAmbient,
-    '4': neutralShadowAmbient,
-    '8': neutralShadowAmbient,
-    '16': neutralShadowAmbient,
-    '28': neutralShadowAmbientDarker,
-    '64': neutralShadowAmbientDarker,
-  };
-  const keyShadow = {
-    '2': { shadowOffset: { width: 0, height: 1 }, shadowOpacity: 0.14, shadowRadius: 2 },
-    '4': { shadowOffset: { width: 0, height: 2 }, shadowOpacity: 0.14, shadowRadius: 4 },
-    '8': { shadowOffset: { width: 0, height: 4 }, shadowOpacity: 0.14, shadowRadius: 8 },
-    '16': { shadowOffset: { width: 0, height: 8 }, shadowOpacity: 0.14, shadowRadius: 16 },
-    '28': { shadowOffset: { width: 0, height: 14 }, shadowOpacity: 0.24, shadowRadius: 28 },
-    '64': { shadowOffset: { width: 0, height: 32 }, shadowOpacity: 0.24, shadowRadius: 64 },
+import { ShadowProps, ShadowTokens, shadowName } from './Shadow.types';
+import { compressible, buildUseTokens, useFluentTheme, mergeProps, UseTokens } from '@fluentui-react-native/framework';
+import { shadowStyleFromTheme } from './shadowStyle';
+
+const useShadowTokens = buildUseTokens<ShadowTokens>(() => ({}), shadowName);
+
+export const Shadow = compressible<ShadowProps, ShadowTokens>((props: ShadowProps, useTokens: UseTokens<ShadowTokens>) => {
+  const { depth } = props;
+  const theme = useFluentTheme();
+
+  const [tokens, cache] = useTokens(theme);
+
+  const [keyShadowTokenStyle] = cache(() => {
+    const keyShadowStyle = shadowStyleFromTheme(theme, depth).key;
+    return {
+      shadowColor: keyShadowStyle.shadowColor,
+      shadowOpacity: keyShadowStyle.shadowOpacity,
+      shadowRadius: keyShadowStyle.shadowRadius,
+      shadowOffset: keyShadowStyle.shadowOffset,
+    };
+  }, ['shadowColor', 'shadowOpacity', 'shadowRadius', 'shadowOffset']);
```

Comment:
Yes sorry I should have commented earlier - talked though this with Saad yesterday and have a lot of local changes (including the fix for this), just working through right now separating them out into commits. I realized there were a lot of issues with the cache (ex. in the screenshot I posted the depth ramp is actually missing one of the shadows). We ended up thinking of using StagedComponent (with memoize) instead of compressible

Reply:
okay sounds good. let me know when i should review, i'll hold off til then.

Reply:
Thanks Ruriko, I pushed some fixes to the cache but there are still known issues and I don't think this PR is ready to review until those are fixed. I've decided to pivot to prototype stagedComponent so will mark this PR as a draft for now.